translator/translate/otel/receiver/prometheus/translator.go

```diff
@@ -0,0 +1,103 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package prometheus
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/aws/amazon-cloudwatch-agent/translator/util"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/receiver"
+	"gopkg.in/yaml.v3"
+
+	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/common"
+)
+
+const (
+	otelConfigParsingError = "has invalid keys: global"
+	defaultTlsCaPath       = "/etc/amazon-cloudwatch-observability-agent-cert/tls-ca.crt"
+)
+
+var (
+	configPathKey = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.PrometheusKey, common.PrometheusConfigPathKey)
+)
+
+type translator struct {
+	name    string
+	factory receiver.Factory
+}
+
+type Option func(any)
+
+var _ common.Translator[component.Config] = (*translator)(nil)
+
+func NewTranslator(opts ...Option) common.Translator[component.Config] {
+	t := &translator{factory: prometheusreceiver.NewFactory()}
+	for _, opt := range opts {
+		opt(t)
+	}
+	return t
+}
+
+func (t *translator) ID() component.ID {
+	return component.NewIDWithName(t.factory.Type(), t.name)
+}
+
+func (t *translator) Translate(conf *confmap.Conf) (component.Config, error) {
+	cfg := t.factory.CreateDefaultConfig().(*prometheusreceiver.Config)
+
+	if !conf.IsSet(configPathKey) {
+		return nil, &common.MissingKeyError{ID: t.ID(), JsonKey: fmt.Sprint(configPathKey)}
```

Comment:
nit: No need to use `fmt.Sprint`. It's already a string.
```suggestion
		return nil, &common.MissingKeyError{ID: t.ID(), JsonKey: configPathKey}
```